shell/platform/linux/fl_platform_views_plugin.cc

```diff
@@ -0,0 +1,305 @@
+static constexpr char kExitMethod[] = "exit";
+
+struct _FlPlatformViewsPlugin {
+  GObject parent_instance;
+
+  FlMethodChannel* channel;
+
+  GHashTable* factories;  // string -> FlPlatformViewFactory*
+
+  GHashTable* platform_views;  // int -> FlPlatformView*
+};
+
+G_DEFINE_TYPE(FlPlatformViewsPlugin, fl_platform_views_plugin, G_TYPE_OBJECT)
+
+// Sends the method call response to Flutter Framework.
+static void send_response(FlMethodCall* method_call,
+                          FlMethodResponse* response) {
+  g_autoptr(GError) error = nullptr;
+  if (!fl_method_call_respond(method_call, response, &error)) {
+    g_warning("Failed to send method call response: %s", error->message);
+  }
+}
+
+// Creates a new platform view.
+static FlMethodResponse* platform_views_create(FlPlatformViewsPlugin* self,
+                                               FlValue* args) {
+  if (fl_value_get_type(args) != FL_VALUE_TYPE_MAP) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Argument map missing or malformed", nullptr));
+  }
+
+  FlValue* id_value = fl_value_lookup_string(args, "id");
+  if (id_value == nullptr || fl_value_get_type(id_value) != FL_VALUE_TYPE_INT) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Missing platform view id", nullptr));
+  }
+  int64_t id = fl_value_get_int(id_value);
+
+  FlValue* view_type_value = fl_value_lookup_string(args, "viewType");
+  if (view_type_value == nullptr ||
+      fl_value_get_type(view_type_value) != FL_VALUE_TYPE_STRING) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Missing platform view viewType", nullptr));
+  }
+  const gchar* view_type = fl_value_get_string(view_type_value);
+
+  FlValue* direction_value = fl_value_lookup_string(args, "direction");
+  if (direction_value == nullptr ||
+      fl_value_get_type(direction_value) != FL_VALUE_TYPE_INT) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Missing platform view direction", nullptr));
+  }
+  GtkTextDirection direction =
+      static_cast<GtkTextDirection>(fl_value_get_int(direction_value));
+
+  FlValue* params_value = fl_value_lookup_string(args, "params");
+
+  FlPlatformView* platform_view =
+      fl_platform_views_plugin_get_platform_view(self, id);
+  if (platform_view) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Platform view id already exists", nullptr));
+  }
+
+  FlPlatformViewFactory* factory =
+      FL_PLATFORM_VIEW_FACTORY(g_hash_table_lookup(self->factories, view_type));
+  if (!factory) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "View type does not exist", nullptr));
+  }
+
+  g_autoptr(FlMessageCodec) codec =
+      fl_platform_view_factory_get_create_arguments_codec(factory);
+  FlValue* creation_params = nullptr;
+  if (codec && params_value) {
```

Comment:
Should we report an error or warning if `params_value != NULL` and `code == NULL`? i.e. it currently would silently lose the parameters. I'm really wondering if it's worth the effort having the codec requested here - why not pass the parameters as a GBytes to `fl_platform_view_factory_create_platform_view` and let the factory work out how to decode them? What do the other Flutter implementations do?

Reply:
iOS implementation does not print any logs when `args[@"params"]` (`params_value` in Linux here) is null.
https://github.com/flutter/engine/blob/2550df40bd9a8825e9b7b46d071c35faf2748f4e/shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm#L149-L156

Android implementation does the same.
https://github.com/flutter/engine/blob/2550df40bd9a8825e9b7b46d071c35faf2748f4e/shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java#L146-L149